storage/store.go

```diff
@@ -1430,20 +1430,69 @@ func (s *Store) WaitForRangeScanCompletion() int64 {
 // updateStoreStatus updates the store's status proto.
 func (s *Store) updateStoreStatus() {
 	now := s.ctx.Clock.Now().WallTime
+	timestamp := proto.Timestamp{WallTime: now}
 	scannerStats := s.scanner.Stats()
 
+	// Get the zone configs.
+	zoneMap, err := s.Gossip().GetInfo(gossip.KeyConfigZone)
+	if err != nil || zoneMap == nil {
+		log.Error("unable to get zone configs")
+		return
+	}
+
+	// Get the leader count and replication count.
+	// TODO(bram): consider moving this to be part of the range scanner directly.
+	var leaderRangeCount, replicatedRangeCount, availableRangeCount int32
+	s.mu.Lock()
+	for raftID, rng := range s.ranges {
+		zoneConfig := zoneMap.(PrefixConfigMap).MatchByPrefix(rng.Desc().StartKey).Config.(*proto.ZoneConfig)
+		raftStatus := s.RaftStatus(raftID)
+		if raftStatus == nil {
+			continue
+		}
+		if raftStatus.SoftState.RaftState == raft.StateLeader {
+			leaderRangeCount++
+			if len(raftStatus.Progress) == len(zoneConfig.ReplicaAttrs) {
```

Comment:
Make this `>=`. Add a TODO to compare attributes of the stores so we can track ranges that have enough replicas but still need to be migrated onto nodes with the desired attributes.